### DISCUSSION TITLE
en-us support removed?

### DISCUSSION THREAD

**Discussion by dumpfheimer** [pndurette/gTTS#276] [Q&A] [answered]
Hi!

I am using gTTS over home assistant.
My announcements stopped working today (and were fixed by manually upgrading to gTTS 2.2.2) but my english announcements are now in british english.
If I understand the code correctly, en-us is falling back to en.

May I ask why en-us support is deprecated?
Is this coming from Google or gTTS?

Best
CH

Similar Home Aassistant issue reference (that resulted in 2.2.2 upgrade):
https://github.com/home-assistant/core/issues/46107

**Comment by pndurette** [comment]
Hi! 

It's unfortunately an upstream change from Google. These 'hyphenated' language codes were undocumented but they used to work, not anymore. 

One hypothesis right now is that 'accent' depends on the geolocation of the request. But there's still lots of unknown.

**Reply by deddc23efb** [reply]
The country specific URLs for translate do result in different accents.  I'm in Canada and don't want the US accent - I prefer British.  It would be great if we could specify the Google Translate base URL : translate.google.com vs translate.google.co.uk as an example.
Edit.  Looking through the code i see the tld is a configurable argument.   I'm coming here from homeassistant - I wonder if they expose that config  in their plugin.

**Reply by dumpfheimer** [reply]
@pndurette thanks for answering so quickly

@deddc23efb 
By looking at the code I would guess it is not possible (yet). The only params passed to gtts are message and language.

`tts = gTTS(text=message, lang=language)`

**Reply by deddc23efb** [reply]
You are correct.  If you're impatient, you can clone the google_translate component in homeassistant and install it as a custom_component.  As a quick hack I modified that line to the following:

`        topleveldomain = "co.uk"`
`        tts = gTTS(text=message, lang=language, tld=topleveldomain)`

and I get my British accented voice back.

**Reply by pndurette** [reply]
> By looking at the code I would guess it is not possible (yet). The only params passed to gtts are message and language.
> 
> `tts = gTTS(text=message, lang=language)`

Just to clarify for all, this is possible in gTTS just not in Home Assistant. So it would have to be implemented over there. I could have a look.

But this is interesting, I wasn't sure it worked. I was able to make it work with a few `co.uk,` `com.au`. It would be nice to list those that work for any language and add them to the docs.

**Comment by pndurette** [comment, marked as answer]
I've added updated documentation on how to obtain local _accents_ with the combination of the right language code (`lang`) and top-level domain (`tld`)! See: https://gtts.readthedocs.io/en/latest/module.html#localized-accents

**Comment by dumpfheimer** [comment]
Very cool!
Do you think it would be a good idea to automatically choose the tld according to the language code provided as parameter?

**Reply by pndurette** [reply]
I prefer to let that to the users. It would be really a chore to find out all the possible casses but also very hard and tedious to maintain imho. 

On the other end, it would indeed provide a nice abstraction, but I don't think the effort is worth it.
